// keystone-admin/client/fields/types.js
/**
 * Helper object of all field components
 */

module.exports = {
	azurefile: require('keystone/fields/admin/azurefile/AzureFileField'),
	boolean: require('keystone/fields/admin/boolean/BooleanField'),
	cloudinaryimage: require('keystone/fields/admin/cloudinaryimage/CloudinaryImageField'),
	cloudinaryimages: require('keystone/fields/admin/cloudinaryimages/CloudinaryImagesField'),
	code: require('keystone/fields/admin/code/CodeField'),
	color: require('keystone/fields/admin/color/ColorField'),
	date: require('keystone/fields/admin/date/DateField'),
	datearray: require('keystone/fields/admin/datearray/DateArrayField'),
	datetime: require('keystone/fields/admin/datetime/DatetimeField'),
	email: require('keystone/fields/admin/email/EmailField'),
	embedly: require('keystone/fields/admin/embedly/EmbedlyField'),
	geopoint: require('keystone/fields/admin/geopoint/GeoPointField'),
	html: require('keystone/fields/admin/html/HtmlField'),
	key: require('keystone/fields/admin/key/KeyField'),
	localfile: require('keystone/fields/admin/localfile/LocalFileField'),
	localfiles: require('keystone/fields/admin/localfiles/LocalFilesField'),
	location: require('keystone/fields/admin/location/LocationField'),
	markdown: require('keystone/fields/admin/markdown/MarkdownField'),
	money: require('keystone/fields/admin/money/MoneyField'),
	name: require('keystone/fields/admin/name/NameField'),
	number: require('keystone/fields/admin/number/NumberField'),
	numberarray: require('keystone/fields/admin/numberarray/NumberArrayField'),
	password: require('keystone/fields/admin/password/PasswordField'),
	relationship: require('keystone/fields/admin/relationship/RelationshipField'),
	s3file: require('keystone/fields/admin/s3file/S3FileField'),
	select: require('keystone/fields/admin/select/SelectField'),
	text: require('keystone/fields/admin/text/TextField'),
	textarea: require('keystone/fields/admin/textarea/TextareaField'),
	textarray: require('keystone/fields/admin/textarray/TextArrayField'),
	url: require('keystone/fields/admin/url/UrlField'),
};
